shreya_custom_loss_test.py

```diff
@@ -0,0 +1,73 @@
+from os import remove
+from tkinter.font import names
+
+# from numpy import error_message
+
+from ansys.motorcad.core import MotorCAD
+
+mc = MotorCAD(open_new_instance=False)
+
+# mc.do_magnetic_calculation()
+
+def add_internal_custom_loss(name, function, type, thermal_node):
+    if type not in ["Electrical", "Mechanical"]:
+        raise ValueError("Thermal Loss Type must be Electrical or Mechanical")
+    else:
+        # Find number of losses
+        no_internal_losses = mc.get_variable("NumCustomLossesInternal_Lab")
+        # Increase array length by 1
+        mc.set_variable("NumCustomLossesInternal_Lab", no_internal_losses + 1)
+        # Define properties
+        mc.set_array_variable("CustomLoss_name_internal_lab", no_internal_losses, name)
+        mc.set_array_variable("CustomLoss_Function_Internal_Lab", no_internal_losses, function)
+        mc.set_array_variable("CustomLoss_Type_Internal_Lab", no_internal_losses, type)
+        mc.set_array_variable("CustomLoss_ThermalNode_Internal_Lab", no_internal_losses, thermal_node)
+
+def add_external_custom_loss(name, power_function, voltage_function):
+    no_external_losses = mc.get_variable("NumCustomLossesExternal_Lab")
+    mc.set_variable("NumCustomLossesExternal_Lab", no_external_losses + 1)
+    mc.set_array_variable('CustomLoss_Name_External_Lab', no_external_losses, name)
+    mc.set_array_variable('CustomLoss_PowerFunction_External_Lab', no_external_losses, power_function)
+    mc.set_array_variable('CustomLoss_VoltageFunction_External_Lab', no_external_losses, voltage_function)
+
+def remove_internal_custom_loss(index):
```

Comment:
Suggestion to reduce code duplication. Something like this?

Left the body empty so you can try and finish it.

```python
# preceding underscore makes it clear to user that this is a function they shouldn't be using ( private method
# https://www.datacamp.com/tutorial/python-private-methods-explained
def _motorcad_array_pop(index, var_length_array, list_of_var_names):
    no__losses = mc.get_variable(var_length_array)

    # Loop through each item in list_of_var_names and remove from array

    mc.set_variable(var_length_array, no__losses - 1)

# useage example
# [] is a list which we can iterate through and can be different lengths
motorcad_array_pop(1,
                   "NumCustomLossesExternal_Lab",
                   ["CustomLoss_Name_External_Lab","CustomLoss_PowerFunction_External_Lab", "CustomLoss_VoltageFunction_External_Lab"])
```